### PR TITLE
Add crate command

### PR DESCRIPTION
**Pull request by gc**
lets you see information on a cargo crate
discord_2019-01-05_07-01-03

### CODE REVIEW

File: commands/Tools/crate.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const getBytes = (bytes) => {
+	const i = Math.floor(Math.log(bytes) / Math.log(1024));
+	return (!bytes && '0 Bytes') || `${(bytes / Math.pow(1024, i)).toFixed(2)} ${suffixes[i]}`;
+};
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, {
+			description: 'Shows the install/publish size of a npm package.',
```
Comment (vladfrangu) [inline commands/Tools/crate.js]: npm package?

File: commands/Tools/crate.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			usage: '<name:str>'
+		});
+	}
+
+	async run(msg, [name]) {
+		const url = new URL(`https://crates.io/api/v1/crates/${encodeURIComponent(name)}`);
```
Comment (vladfrangu) [inline commands/Tools/crate.js]: Whats the point of the new URL here?

File: commands/Tools/crate.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+		const { crate, versions } = await fetch(url)
+			.then(response => response.json())
+			.catch(() => {
+				throw 'There was an unexpected error. Try again later.';
+			});
+
+		if (!crate) throw 'That crate doesn\'t exist.';
+
+		const latest = versions[0];
```
Comment (vladfrangu) [inline commands/Tools/crate.js]:
```suggestion
		const [latest] = versions;
```

File: commands/Tools/crate.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+const { Command } = require('klasa');
+const fetch = require('node-fetch');
+const { MessageEmbed } = require('discord.js');
+
+const suffixes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
```
Comment (kyranet) [inline commands/Tools/crate.js]: I don't think a crate will be over a gb while our lives still last 😂

File: commands/Tools/crate.js [hunk 5]
```diff
@@ -12,24 +12,22 @@ module.exports = class extends Command {
...
 		super(...args, {
-			description: 'Shows the install/publish size of a npm package.',
+			description: 'Shows the install/publish size of a cargo crate.',
 			usage: '<name:str>'
 		});
 	}
 
 	async run(msg, [name]) {
-		const url = new URL(`https://crates.io/api/v1/crates/${encodeURIComponent(name)}`);
+		const url = `https://crates.io/api/v1/crates/${encodeURIComponent(name)}`;
```
Comment (vladfrangu) [inline commands/Tools/crate.js]: Just move it to the await fetch call w;w

**Pull request merged**